esx_service/utils/kvESX.py

```diff
@@ -175,24 +185,28 @@ def load(volpath):
     except IOError:
         return None
 
-    kvDict = json.load(fh)
-
+    kv_str = fh.read()
     fh.close()
+    try:
+       kv_dict = json.loads(kv_str)
```

Comment:
The try-except is in case the kv_str is a null dict from json.loads. Like
if the meta-data is corrupt.

On Thu, Jun 2, 2016 at 12:44 PM, Mark Sterin <notifications@github.com>
wrote:

> In esx_service/utils/kvESX.py
> <https://github.com/vmware/docker-volume-vsphere/pull/433#discussion_r65493718>
> :
>
> >      fh.close()
> > +    try:
> > +       kv_dict = json.loads(kv_str)
>
> why this try/except is needed ? Does not upstairs need to check for return
> value anyways, so it could haqve handles exceptions instead? I think we
> need to catch exception only when a decision is needed, and up to this
> moment we can simply pass it up (which is default behavior)
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/433/files/1ae935fbbc0c3205389f8fd83888b9afea83cba2#r65493718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APHseCilCRMjTJ4VObfCbGS1TGksDLiQks5qHoLugaJpZM4IrROI>
> .
>